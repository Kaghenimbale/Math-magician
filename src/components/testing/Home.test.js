import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Home from '../Home';

describe('Render components', () => {
  it('It render the Home Components', () => {
    render(<Home />);
    expect(screen.getByText('Welcome to our Page')).toBeInTheDocument();
  });

  test('Remder Home component', () => {
    const tree = renderer
      .create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
