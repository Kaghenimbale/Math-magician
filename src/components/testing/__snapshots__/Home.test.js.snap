// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components Remder Home component 1`] = `
<div
  className="home-content"
>
  <h1>
    Welcome to our Page
  </h1>
  <p>
    Mathematical reasoning requires rigor. This means that the definitions must be absolutely unambiguous and the proofs must be reducible to a succession of applications of inference rules, without any use of empirical evidence and intuition.Rigorous reasoning is not specific to mathematics, but, in mathematics, the standard of rigor is much higher than elsewhere. Despite mathematics concision, rigorous proofs can require hundreds of pages to express. The emergence of computer-assisted proofs has allowed proof lengths to further expand, such as the 255-page Feitâ€“Thompson theorem. The result of this trend is a philosophy of the quasi-empiricist proof that can not be considered infallible, but has a probability attached to it. The concept of rigor in mathematics dates back to ancient Greece, where their society encouraged logical, deductive reasoning. However, this rigorous approach would tend to discourage exploration of new approaches, such as irrational numbers and concepts of infinity. The method of demonstrating rigorous proof was enhanced in the sixteenth century through the use of symbolic notation. In the 18th century, social transition led to mathematicians earning their keep through teaching, which led to more careful thinking about the underlying concepts of mathematics. This produced more rigorous approaches, while transitioning from geometric methods to algebraic and then arithmetic proofs. At the end of the 19th century, it appeared that the definitions of the basic concepts of mathematics were not accurate enough for avoiding paradoxes (non-Euclidean geometries and Weierstrass function) and contradictions. This was solved by the inclusion of axioms with the apodictic inference rules of mathematical theories; the re-introduction of axiomatic method pioneered by the ancient Greeks. It results that is no more a relevant concept in mathematics, as a proof is either correct or erroneous, and a is simply a pleonasm. Where a special concept of rigor comes into play is in the socialized aspects of a proof, wherein it may be demonstrably refuted by other mathematicians. After a proof has been accepted for many years or even decades, it can then be considered as reliable
  </p>
</div>
`;
